llama.cpp.zig/prompt.zig

```diff
@@ -265,6 +265,7 @@ pub fn generateAppendOne(self: *@This()) !Token {
     const new_token = try self.sampling.sample(self.ctx, null, self.batch.n_tokens - 1);
     self.tokens_mutex.unlock();
     try self.addOneToken(new_token);
+    if (self.sampling.grammar != null) self.sampling.accept(self.ctx, new_token, true);
```

Comment:
@Deins Maybe you have better idea for this. As I understand tokens from the prompt have to be accepted without grammar options as `false` and generated tokens should use `true`.

Reply:
Not completely sure. But from my understanding I think it depends on context / use case, maybe even model if its specialized. Standard use case might be where you give system prompt lets say "Translate users request to json". Then you would not accept neither system prompt nor users request and reset the grammar before each generation, to force sampling to generate only valid json.
On other hand for some sort of autocomplete use case where you user gives plain text description and then first half of json and model auto completes it. From grammars point of view all json characters should be accepted so it looks as whole.

At least thats how I understand it.

I think it might be needed that user can specify if grammar should be accepted or not as function argument maybe.
But for now I think these defaults to only accept generated tokens, make sense. 
The prompt should just be helper/utility should to handle most common use case. For anything more specific there is no way around of using the lower level llama functions directly.